qcodes/tests/test_parameter.py

```diff
@@ -1856,3 +2176,200 @@ def test_get_from_cache_does_not_trigger_real_get_if_get_if_invalid_false():
+    assert local_parameter._get_count == 0
+    # now fake the time stamp so get should be triggered
+    set_time = start - timedelta(seconds=10)
+    local_parameter.cache._update_with(
+        value=value, raw_value=value, timestamp=set_time)
+    # now that ts < max_val_age calling get_latest should update the time
+    assert local_parameter.cache.timestamp == set_time
+    assert local_parameter.cache.get() == value
+    assert local_parameter._get_count == 1
+    assert local_parameter.cache.timestamp >= start
+
+
+def test_no_get_max_val_age():
+    """
+    Test that cache.get on a parameter with max_val_age set and
+    no get cmd raises correctly.
+    """
+    value = 1
+    with pytest.raises(SyntaxError):
+        _ = Parameter('test_param', set_cmd=None,
+                      get_cmd=False,
+                      max_val_age=1, initial_value=value)
+
+
+def test_no_get_max_val_age_runtime_error(get_if_invalid):
+    """
+    _BaseParameter does not have a check on creation time that
+    no get_cmd is mixed with max_val_age since get_cmd could be added
+    in a subclass. Here we create a subclass that does not add a get
+    command and also does not implement the check for max_val_age
+    """
+    value = 1
+
+    class LocalParameter(_BaseParameter):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+            self.set_raw = lambda x: x
+            self.set = self._wrap_set(self.set_raw)
+
+    local_parameter = LocalParameter('test_param',
+                                     None,
+                                     max_val_age=1)
+    start = datetime.now()
+    set_time = start - timedelta(seconds=10)
+    local_parameter.cache._update_with(
+        value=value, raw_value=value, timestamp=set_time)
+
+    if get_if_invalid is True:
+        with pytest.raises(RuntimeError, match="max_val_age` is not supported"):
+            local_parameter.cache.get(get_if_invalid=get_if_invalid)
+    elif get_if_invalid == NOT_PASSED:
+        with pytest.raises(RuntimeError, match="max_val_age` is not supported"):
+            local_parameter.cache.get()
+    else:
+        assert local_parameter.cache.get(get_if_invalid=get_if_invalid) == 1
+
+
+def test_no_get_timestamp_none_runtime_error(get_if_invalid):
+    """
+    Test that a parameter that has never been
+    set, cannot be get and does not support
+    getting raises a RuntimeError.
+    """
+    local_parameter = Parameter('test_param', get_cmd=False)
+
+    if get_if_invalid is True:
+        with pytest.raises(RuntimeError, match="Value of parameter test_param"):
+            local_parameter.cache.get(get_if_invalid=get_if_invalid)
+    elif get_if_invalid == NOT_PASSED:
+        with pytest.raises(RuntimeError, match="Value of parameter test_param"):
+            local_parameter.cache.get()
+    else:
+        assert local_parameter.cache.get(get_if_invalid=get_if_invalid) is None
```

Comment:
@astafan8 Maybe we should moveall the new tests to 2 new files. test_parameter_snapshot and test_parameter_cache ? this one is getting a bit extreem

Reply:
@jenshnielsen yes, sure! perhaps it's even time to turn "test_parameter.py" into "parameter" test module

Reply:
Thinking a bit more about this and the conflicts this had in the tests with one of the other prs I will do this in another pr soon